packages/components/forms/src/Multiselect/Multiselect.tsx

```diff
@@ -0,0 +1,286 @@
+   */
+  onSearchValueChange?: (event: React.ChangeEvent) => void;
+
+  /**
+   * This is the value will be passed to the `placeholder` prop of the input.
+   * @default "Search"
+   */
+  searchPlaceholder?: string;
+
+  /**
+   * A message that will be shown when it is not possible to find any option that matches the input value
+   * @default "No matches"
+   */
+  noMatchesMessage?: string;
+
+  /**
+   * Use this prop to get a ref to the input element of the component
+   */
+  searchInputRef?: React.Ref<HTMLInputElement>;
+  /**
+   * Use this prop to get a ref to the toggle button of the component
+   */
+  toggleRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * Use this prop to get a ref to the list of items of the component
+   */
+  listRef?: React.Ref<HTMLUListElement>;
+
+  /**
+   * It sets the width of the list
+   * @default "auto"
+   */
+  listWidth?: 'auto' | 'full';
+
+  /**
+   * It sets the max-height, in pixels, of the list
+   * The default value is the height of 5 single line items
+   * @default 180
+   */
+  listMaxHeight?: number;
+  /**
+   * Sets the list to show its loading state
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * Boolean to control whether or not to render the suggestions box in a React Portal.
+   * Rendering content inside a Portal allows the suggestions box to escape the bounds
+   * of its parent while still being positioned correctly.
+   * Defaults to `false`
+   */
+  usePortal?: boolean;
+}
+
+function _Multiselect(props: MultiselectProps, ref: React.Ref<HTMLDivElement>) {
+  const {
+    id,
+    className,
+    startIcon,
+    placeholder = 'Select one or more Items',
+    currentSelection = [],
+    hasSearch = false,
+    defaultValue = '',
+    onSearchValueChange,
+    searchPlaceholder = 'Search',
+    searchInputRef,
+    noMatchesMessage = 'No matches found',
+    toggleRef,
+    listRef,
+    listWidth = 'auto',
+    listMaxHeight = 180,
+    isLoading = false,
+    usePortal = false,
+    testId = 'cf-multiselect',
+    children,
+  } = props;
+
+  const styles = getMultiselectStyles();
+
+  const [searchValue, setSearchValue] = useState(defaultValue);
+  const [isOpen, setIsOpen] = useState(false);
+
+  const internalSearchInputRef = useRef(null);
+
+  const handleSearchChange = (event) => {
+    setSearchValue(event.target.value);
+    onSearchValueChange(event);
+  };
+
+  const resetSearch = () => {
+    // this looks a bit hacky, but is the official way of externally triggering the onChange handler for an input
+    // https://stackoverflow.com/a/46012210/17269164
+
+    const nativeInputValueSetter = Object.getOwnPropertyDescriptor(
+      window.HTMLInputElement.prototype,
+      'value',
+    ).set;
+    nativeInputValueSetter.call(internalSearchInputRef.current, '');
+    const forcedEvent = new Event('change', { bubbles: true });
+    internalSearchInputRef.current.dispatchEvent(forcedEvent);
+  };
```

Comment:
I'm not sure I understand why we can't do `setSearchValue('')` instead of this, would you mind explaining?

Reply:
If we only clear searchValue, it will not call the callback function onSearchChange with the input change event, so the filter of the list will not be reset.
One big issue we had when using the autocomplete, was that, when the input changed, you never got the full event, but only the value of the input. Which lead down to a list of issues for onChange / onBlur validations.

This way the reset of the search value will trigger the expected onChange event. In the linked stackoverflow thread there is an explanation from Dan Abramov about why programmatically changing the value of an input is not firing the native event handler.